### ISSUE TITLE
Unable to set up project, please provide more specific instructions

### ISSUE THREAD

**Issue by Hokage3211** [OpenMAR/PiTool#3]
There are several problems that I'm not sure if Pimax is aware of with this project, namely, it's using outdated executables and libraries. I made another issue about this, but I figured a new issue may be more fitting. 

Before any of this, they should also mention it requires visual studio to edit, just so it's clear what platform they use to code, for complete transparency.

**1)** The qt version they want to use is not a currently supported version on the website, and using the archives to find it doesn't seem to provide a proper version, but I'm going to be attempting to use the 5.9 version instead, which seems to be supported.
**2)** The visual studio C++ toolset is also out of date, and I'm not sure by how much as it doesn't say, so I'm not sure if that will affect the project development. This is the dialog I get when opening it up in visual studio 
![image](https://user-images.githubusercontent.com/54164358/65567488-4c584b80-df0b-11e9-8004-21aab86f2bae.png)
**3)** "add the bin folder to the environment variable" isn't quite clear, and maybe that's my fault, but I'd like at least to know where they see this variable to "add" to, since what I seem to find is something you set, rather than add to, so maybe I'm wrong. 
I haven't been able to make the project have no errors yet, as it says "can't open file" on many "Q" libraries/dependencies, so I'm having trouble figuring out mainly how to link qt to it properly, and mainly would like to have some better instructions on what to do, as that would help. 

I'm very excited to begin contributing as I very much enjoy my Pimax 5K+, but think small changes would make a world of difference, but I would appreciate a little more help than just "here's the code" and the keywords "qt" and "python" (speaking figuratively on the last part).

I don't pretend I know how to work every single bit of this all, but I think my confusion is genuine at least. 

Thank you for your time.

**Comment by Hokage3211** [comment]
Wait maybe I'm confused, is qt the editor? I see both visual studio and qt files, so I'm unsure what to develop it with, if qt, then we can't get ahold of the proper version they seem to want us to use? 

Opening the project in 5.9.0 is possible, as I've done so, but not sure if that's appropriate given any number of problems that can arise from one-version-off stuff, as many developers know single version number differences can give compatibility changes.

**Comment by Hokage3211** [comment]
Alright to follow up, the qt editor apparently can be found here, thanks to genesis for finding this, and I am working on confirming his strategy to get it to compile. https://download.qt.io/archive/qt/5.8/5.8.0/qt-opensource-windows-x86-msvc2015_64-5.8.0.exe

**Comment by jcp219** [comment]
I was able open PiTool.vcxproj in VS 2017 Community Edition and compile PiTool.exe by doing the following:
- Install QT 5.9
- Install latest Python
- Open PiTool.vcxproj in notepad, find/replace the following:
    - D:\Tools\Qt\Qt5.8.0\5.8 with your QT path, for example C:\Tools\QT\5.9
    - D:/Tools/Qt/Qt5.8.0/5.8 with your QT path, for example C:/Tools/QT/5.9
- Open PiTool.vcxproj.user in notepad, find/replace the following:
    - D:\Tools\Qt\Qt5.8.0\5.8 with your QT path, for example C:\Tools\QT\5.9
- Open PiTool.vcxproj with VS 2017
- Accept the prompt to upgrade to whichever version of the runtime you have installed
- Switch to Release mode
- Right-Click the solution and build

This builds PiTool.exe but not PiSvc.dll, still working on that.  In the meantime you can copy PiSvc.dll from PiTool\piSvc\ to your PiTool\release folder so you can run your newly compiled PiTool.exe.

Hope this is helpful.

**Comment by NGenesis** [comment]
There are the steps I took to get it working:

- Download and install the latest version of `Python 3` from https://www.python.org/downloads/
- Download and install `Qt 5.8.0 for MSVC2015` from https://download.qt.io/archive/qt/5.8/5.8.0/qt-opensource-windows-x86-msvc2015_64-5.8.0.exe
- Download and install `Visual Studio 2015 Community` from https://go.microsoft.com/fwlink/?LinkId=532606&clcid=0x409

For each build session:
- Run `Qt 5.8 64-bit for Desktop (MSVC 2015)` from the Start Menu.
- Run `"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" amd64` from the Command Prompt window.
- Run `X:\path\to\pitool\repo\build_pack_tool.bat` from the Command Prompt window.
- Type in a version (e.g. `1.0.1.180`) when prompted.

This will generate a `setup` folder with the PiTool installer, and an `out` folder with the individual binaries in the root of the repository.

**Comment by Hokage3211** [comment]
As a follow up, I have been able to get it to build using the steps provided by genesis one post above, some things to note in case anyone has any trouble; 
1) I ran the amd64 thing from my 2017 install, so look through your other versions if the vcvarsall doesn't show up where he says it is.
2) I had to make sure to install the c++ stuff on the 2015 community version, which it didn't do automatically, along with making sure to install the 8.1 SDK (which should be installed along with the c++ toolset/libraries/stuff) 
3) QT is the approved editor as far as I can tell, as opening the project in QT doesn't affect the files, while opening it in visual studio alters some files, hinting which program it's supposed to load into.
4) You do actually need to have the Community 2015 version, sadly, my 2017 community/professional (I have both) installations didn't let me compile it
5) adding "environment variables" for anyone who hasn't seen them before means something like this; https://geek-university.com/python/add-python-to-the-windows-path/ and users may want to also add the bin folder there, as well as python installation, and the chosen msbuild.exe version they can find in their files, (if there is two, I chose the one under the AMD-something- file folder)

I think that covers my problems and any I could think of, hopefully this will help more users.

Now I just need to look over the code and see if I can get the UI designer to work, but that's a separate issue. Thank you all for your help, and thank you very much genesis for your talent in this.
